### ISSUE TITLE
dumpconfig creates configuration with wrong chain

### ISSUE THREAD

**Issue by istoian-ledger** [ethereum/go-ethereum#23112]
#### System information

Geth version: `1.10.4`
OS & Version: Ubuntu 20.04.2 LTS
Commit hash : aa637fd3

#### Expected behaviour
Node is started with the correct configuration
````
  protocols: {
    eth: {
      config: {
        berlinBlock: 9812189,
        byzantiumBlock: 1700000,
        chainId: 3,
        constantinopleBlock: 4230000,
        daoForkSupport: true,
        eip150Block: 0,
        eip150Hash: "0x41941023680923e0fe4d74a34bdac8141f2540e3ae90623718e47d66d1ca4a2d",
        eip155Block: 10,
        eip158Block: 10,
        ethash: {},
        homesteadBlock: 0,
        istanbulBlock: 6485846,
        londonBlock: 10499401,
        muirGlacierBlock: 7117117,
        petersburgBlock: 4939394
      },
      difficulty: 636744809587392,
      genesis: "0x41941023680923e0fe4d74a34bdac8141f2540e3ae90623718e47d66d1ca4a2d",
      head: "0x6c32de349a0d9579e7e9e28e3e93329e0c4c501d205ef4dc1af03b0a3b582866",
      network: 3
    },
    snap: {}
  }
````

#### Actual behaviour
Node starts with incorrect configuration
````
  protocols: {
    eth: {
      config: {
        berlinBlock: 12244000,
        byzantiumBlock: 4370000,
        chainId: 1,
        constantinopleBlock: 7280000,
        daoForkBlock: 1920000,
        daoForkSupport: true,
        eip150Block: 2463000,
        eip150Hash: "0x2086799aeebeae135c246c65021c82b4e15a2c451340993aacfd2751886514f0",
        eip155Block: 2675000,
        eip158Block: 2675000,
        ethash: {},
        homesteadBlock: 1150000,
        istanbulBlock: 9069000,
        muirGlacierBlock: 9200000,
        petersburgBlock: 7280000
      },
      difficulty: 17179869184,
      genesis: "0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3",
      head: "0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3",
      network: 3
    },
    snap: {}
````

#### Steps to reproduce the behaviour
When bootstrapping geth node in ropsten chain with 
`/home/ethereum/ropsten/bin/1.10.4/geth --ropsten --syncmode full --gcmode archive --http --http.port 20100 --port 20150 --cache 4096 --http.api admin,net,eth,txpool,web3,debug --http.vhosts "*" --datadir /home/ethereum/ropsten/data dumpconfig > /home/ethereum/ropsten/configs/ethereum-ropsten.conf`

and later starting the node with 
`/home/ethereum/ropsten/bin/1.10.4/geth -config /home/ethereum/ropsten/configs/ethereum-ropsten.conf -datadir /home/ethereum/ropsten/data`

the node is not able to sync with the chain due to the absence of peers.

However, if running node with 
`/home/ethereum/ropsten/bin/1.10.4/geth --ropsten --syncmode full --gcmode archive --http --http.port 20100 --port 20150 --cache 4096 --http.api admin,net,eth,txpool,web3,debug --http.vhosts "*" --datadir /home/ethereum/ropsten/data ` and letting it run, it finds peers and starts syncing with chain. After this, dumping config file and using it for consequent starts of the node works flawlessly.

#### Backtrace

````
Starting Ethereum Ropsten node (geth)...
Started Ethereum Ropsten node (geth).
INFO [06-25|15:04:06.136] Starting Geth on Ethereum mainnet...
INFO [06-25|15:04:06.136] Bumping default cache on mainnet         provided=1024 updated=4096
INFO [06-25|15:04:06.138] Maximum peer count                       ETH=50 LES=0 total=50
INFO [06-25|15:04:06.138] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [06-25|15:04:06.138] Enabling recording of key preimages since archive mode is used
INFO [06-25|15:04:06.138] Set global gas cap                       cap=50,000,000
INFO [06-25|15:04:06.138] Allocated trie memory caches             clean=1.20GiB dirty=0.00B
INFO [06-25|15:04:06.138] Allocated cache and file handles         database=/home/ethereum/ropsten/data/geth/chaindata cache=2.00GiB handles=262,144
INFO [06-25|15:04:06.145] Opened ancient database                  database=/home/ethereum/ropsten/data/geth/chaindata/ancient readonly=false
INFO [06-25|15:04:06.145] Writing default main-net genesis block
INFO [06-25|15:04:06.259] Persisted trie from memory database      nodes=12356 size=1.78MiB time=32.840972ms gcnodes=0 gcsize=0.00B gctime=0s livenodes=1 livesize=0.00B
INFO [06-25|15:04:06.260] Initialised chain configuration          config="{ChainID: 1 Homestead: 1150000 DAO: 1920000 DAOSupport: true EIP150: 2463000 EIP155: 2675000 EIP158: 2675000 Byzantium: 4370000 Constantinople: 7280000 Petersburg: 7280000 Istanbul: 9069000, Muir Glacier: 9200000, Berlin: 12244000, London: <nil>, Engine: ethash}"
INFO [06-25|15:04:06.260] Disk storage enabled for ethash caches   dir=/home/ethereum/ropsten/data/geth/ethash count=3
INFO [06-25|15:04:06.260] Disk storage enabled for ethash DAGs     dir=/home/ethereum/.ethash                  count=2
INFO [06-25|15:04:06.260] Initialising Ethereum protocol           network=3 dbversion=<nil>
INFO [06-25|15:04:06.260] Loaded most recent local header          number=0 hash=d4e567..cb8fa3 td=17,179,869,184 age=52y2mo3w
INFO [06-25|15:04:06.260] Loaded most recent local full block      number=0 hash=d4e567..cb8fa3 td=17,179,869,184 age=52y2mo3w
INFO [06-25|15:04:06.260] Loaded most recent local fast block      number=0 hash=d4e567..cb8fa3 td=17,179,869,184 age=52y2mo3w
WARN [06-25|15:04:06.260] Failed to load snapshot, regenerating    err="missing or corrupted snapshot"
INFO [06-25|15:04:06.260] Rebuilding state snapshot
INFO [06-25|15:04:06.261] Resuming state snapshot generation       root=d7f897..0f0544 accounts=0 slots=0 storage=0.00B elapsed="325.845µs"
INFO [06-25|15:04:06.261] Regenerated local transaction journal    transactions=0 accounts=0
INFO [06-25|15:04:06.261] Gasprice oracle is ignoring threshold set threshold=2
WARN [06-25|15:04:06.261] Error reading unclean shutdown markers   error="leveldb: not found"
INFO [06-25|15:04:06.261] Starting peer-to-peer node               instance=Geth/v1.10.4-stable-aa637fd3/linux-amd64/go1.16.4
INFO [06-25|15:04:06.263] New local node record                    seq=1 id=769054eac948d1e2 ip=127.0.0.1 udp=20150 tcp=20150
INFO [06-25|15:04:06.263] Started P2P networking                   self=enode://34940c98f1b13fb9a50235c163eeebe888f5339a8c5668c406410c428de8d9f0442e74edd2e8ea77cb560e4b348311df9725deb8698fa3db3902aea1ed2fabd2@127.0.0.1:20150
INFO [06-25|15:04:06.264] IPC endpoint opened                      url=/home/ethereum/ropsten/data/geth.ipc
INFO [06-25|15:04:06.264] HTTP server started                      endpoint=10.99.0.236:20100 prefix= cors= vhosts=*
INFO [06-25|15:04:06.330] Generated state snapshot                 accounts=8893 slots=0 storage=409.64KiB elapsed=69.706ms
INFO [06-25|15:04:12.055] New local node record                    seq=2 id=769054eac948d1e2 ip=51.178.91.200 udp=20150 tcp=20150
INFO [06-25|15:04:16.322] Looking for peers                        peercount=0 tried=15 static=0
WARN [06-25|15:04:26.082] Served eth_coinbase                      conn=10.99.0.236:45854 reqid=3 t="20.269µs" err="etherbase must be explicitly specified"
INFO [06-25|15:04:26.445] Looking for peers                        peercount=0 tried=25 static=0
INFO [06-25|15:04:36.446] Looking for peers                        peercount=0 tried=27 static=0
INFO [06-25|15:04:46.555] Looking for peers                        peercount=0 tried=29 static=0
INFO [06-25|15:04:56.868] Looking for peers                        peercount=0 tried=38 static=0
INFO [06-25|15:05:06.951] Looking for peers                        peercount=0 tried=34 static=0
INFO [06-25|15:05:17.110] Looking for peers                        peercount=0 tried=30 static=0
INFO [06-25|15:05:27.282] Looking for peers                        peercount=0 tried=33 static=0
INFO [06-25|15:05:37.305] Looking for peers                        peercount=0 tried=33 static=0
INFO [06-25|15:05:47.563] Looking for peers                        peercount=0 tried=30 static=0

````

**Comment by s1na** [comment]
I tried to reproduce this. Doing `./geth --ropsten --config ropsten.conf` works, but as op suggests `./geth --config ropsten.conf` doesn't seem to suffice and geth generates the mainnet genesis and uses mainnet chainId. Will dig a bit deeper.

**Comment by s1na** [comment]
I asked others about this and found out that the chain flag needs to be provided along with `--config` and documented that in #23146. As far as I gathered, the reason is not all chain info is stored in the config file and that's a deliberate design decision to avoid difficulties for future hardforks.

So even though this is a sub-optimal user experience it'll probably stay this way at least until a relatively significant re-design. Closing for now. Thanks for reporting it.
